### PR TITLE
[BEAM-852] Add validation to file based sources during create time

### PR DESCRIPTION
**Pull request by sb2nov**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:
- [x] Make sure the PR title is formatted like:
  `[BEAM-<Jira issue #>] Description of pull request`
- [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
     Travis-CI on your fork and ensure the whole test matrix passes).
- [x] Replace `<Jira issue #>` in the title with the actual Jira issue
     number, if there is one.
- [ ] If this contribution is large, please file an Apache
     [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

Adding validation to the fileSourceIO so that we can inform the user if the files don't exist at the create time. There is a flag to bypass verification incase your local system doesn't have the same access as the runner etc. 

PTAL @chamikaramj

### CODE REVIEW

File: sdks/python/apache_beam/io/filebasedsource.py [hunk 1]
```diff
@@ -43,7 +43,8 @@ def __init__(self,
                file_pattern,
                min_bundle_size=0,
                compression_type=fileio.CompressionTypes.AUTO,
-               splittable=True):
+               splittable=True,
+               disable_validation=False):
```
Comment (chamikaramj) [inline sdks/python/apache_beam/io/filebasedsource.py]: Let's change this to 'validate=True'

File: sdks/python/apache_beam/io/filebasedsource.py [hunk 2]
```diff
@@ -84,6 +85,9 @@ def __init__(self,
     else:
       # We can't split compressed files efficiently so turn off splitting.
       self._splittable = False
+    if not disable_validation and not self.validate():
+      raise IOError(
```
Comment (chamikaramj) [inline sdks/python/apache_beam/io/filebasedsource.py]: This error should be raised from method 'validate()'.

File: sdks/python/apache_beam/io/filebasedsource.py [hunk 3]
```diff
@@ -134,6 +137,11 @@ def _estimate_sizes_in_parallel(file_names):
       finally:
         pool.terminate()
 
+  def validate(self):
```
Comment (chamikaramj) [inline sdks/python/apache_beam/io/filebasedsource.py]: Please change to '_validate(self)'

File: sdks/python/apache_beam/io/textio.py [hunk 4]
```diff
@@ -72,9 +72,10 @@ def position(self, value):
 
   def __init__(self, file_pattern, min_bundle_size,
                compression_type, strip_trailing_newlines, coder,
-               buffer_size=DEFAULT_READ_BUFFER_SIZE):
+               buffer_size=DEFAULT_READ_BUFFER_SIZE, disable_validation=False):
```
Comment (chamikaramj) [inline sdks/python/apache_beam/io/textio.py]: Please add this option to Avro source (avroio.py) as well.

Comment (chamikaramj) [inline sdks/python/apache_beam/io/textio.py]: validate=True

File: sdks/python/apache_beam/io/filebasedsource.py [hunk 5]
```diff
@@ -43,7 +43,8 @@ def __init__(self,
                file_pattern,
                min_bundle_size=0,
                compression_type=fileio.CompressionTypes.AUTO,
-               splittable=True):
+               splittable=True,
+               disable_validation=False):
     """Initializes ``FileBasedSource``.
 
     Args:
```
Comment (chamikaramj) [inline sdks/python/apache_beam/io/filebasedsource.py]: Please update Args list here.

File: sdks/python/apache_beam/io/textio.py [hunk 6]
```diff
@@ -72,9 +72,10 @@ def position(self, value):
...
     super(_TextSource, self).__init__(file_pattern, min_bundle_size,
-                                      compression_type=compression_type)
+                                      compression_type=compression_type,
+                                      disable_validation=disable_validation)
 
     self._strip_trailing_newlines = strip_trailing_newlines
     self._compression_type = compression_type
```
Comment (chamikaramj) [inline sdks/python/apache_beam/io/textio.py]: You have to add this option to 'ReadFromText' transform and pass it to _TextSource (and update ReadFromAvro transform similarly).

**Comment by sb2nov** [comment]
Thanks for the comments

R: @chamikaramj

**Comment by sb2nov** [comment]
retest this please

**Comment by sb2nov** [comment]
R: @chamikaramj updated the documentation now.

**Comment by chamikaramj** [comment]
LGTM

**Comment by sb2nov** [comment]
R: @robertwb

**Comment by sb2nov** [comment]
Ping @robertwb for merging

**Comment by sb2nov** [comment]
R: @robertwb Please merge

**Comment by robertwb** [comment]
Will merge as soon as I verify the tests are all passing.

**Comment by sb2nov** [comment]
Thanks

**Pull request closed without merging**